precompiles/staking/method_distribute_test.go

```diff
@@ -0,0 +1,278 @@
+		require.False(t, ok)
+
+		// End fee collector balance should be 0.
+		balance, err := s.sdkKeepers.BankKeeper.Balance(s.ctx, &banktypes.QueryBalanceRequest{
+			Address: feeCollectorAddress,
+			Denom:   zrc20Denom,
+		})
+		require.NoError(t, err)
+		require.Equal(t, uint64(0), balance.Balance.Amount.Uint64())
+	})
+
+	t.Run("should fail to distribute more than allowed to staking", func(t *testing.T) {
+		// Setup test.
+		s := newTestSuite(t)
+		zrc20Denom := precompiletypes.ZRC20ToCosmosDenom(s.zrc20Address)
+
+		// Set caller balance.
+		s.fungibleKeeper.DepositZRC20(s.ctx, s.zrc20Address, s.defaultCaller, big.NewInt(1000))
+
+		// Allow staking to spend ZRC20 tokens.
+		allowStaking(t, s, big.NewInt(999))
+
+		// Setup method input.
+		s.mockVMContract.Input = packInputArgs(
+			t,
+			s.methodID,
+			[]interface{}{s.zrc20Address, big.NewInt(1000)}...,
+		)
+
+		// Call method.
+		success, err := s.contract.Run(s.mockEVM, s.mockVMContract, false)
+
+		// Check error.
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "invalid allowance, got 999, wanted 1000")
+
+		// Unpack and check result boolean.
+		res, err := s.methodID.Outputs.Unpack(success)
+		require.NoError(t, err)
+
+		ok := res[0].(bool)
+		require.False(t, ok)
+
+		// End fee collector balance should be 0.
+		balance, err := s.sdkKeepers.BankKeeper.Balance(s.ctx, &banktypes.QueryBalanceRequest{
+			Address: feeCollectorAddress,
+			Denom:   zrc20Denom,
+		})
+		require.NoError(t, err)
+		require.Equal(t, uint64(0), balance.Balance.Amount.Uint64())
+	})
+
+	t.Run("should fail to distribute more than user balance", func(t *testing.T) {
+		// Setup test.
+		s := newTestSuite(t)
+		zrc20Denom := precompiletypes.ZRC20ToCosmosDenom(s.zrc20Address)
+
+		// Set caller balance.
+		s.fungibleKeeper.DepositZRC20(s.ctx, s.zrc20Address, s.defaultCaller, big.NewInt(1000))
+
+		// Allow staking to spend ZRC20 tokens.
+		allowStaking(t, s, big.NewInt(100000))
+
+		// Setup method input.
+		s.mockVMContract.Input = packInputArgs(
+			t,
+			s.methodID,
+			[]interface{}{s.zrc20Address, big.NewInt(1001)}...,
+		)
+
+		success, err := s.contract.Run(s.mockEVM, s.mockVMContract, false)
+
+		// Check error.
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "execution reverted")
+
+		// Unpack and check result boolean.
+		res, err := s.methodID.Outputs.Unpack(success)
+		require.NoError(t, err)
+
+		ok := res[0].(bool)
+		require.False(t, ok)
+
+		// End fee collector balance should be 0.
+		balance, err := s.sdkKeepers.BankKeeper.Balance(s.ctx, &banktypes.QueryBalanceRequest{
+			Address: feeCollectorAddress,
+			Denom:   zrc20Denom,
+		})
+		require.NoError(t, err)
+		require.Equal(t, uint64(0), balance.Balance.Amount.Uint64())
+	})
+
+	t.Run("should distribute and lock ZRC20", func(t *testing.T) {
+		// Setup test.
+		s := newTestSuite(t)
+
+		// Set caller balance.
+		s.fungibleKeeper.DepositZRC20(s.ctx, s.zrc20Address, s.defaultCaller, big.NewInt(1000))
```

Comment:
```suggestion
	_, err := s.fungibleKeeper.DepositZRC20(s.ctx, s.zrc20Address, s.defaultCaller, big.NewInt(1000))
		require.NoError(t, err)
```

Reply:
Added in f206701e3453bfa9750518d3ba81597fec38c9f9